### ISSUE TITLE
npm install lightdash fails without unixODBC already installed, perhaps include it into npm install lightdash?

### ISSUE THREAD

**Issue by djayatillake** [lightdash/lightdash#3361]
### Description

npm install lightdash fails if you don't have unixODBC already installed.

### To Reproduce

`npm install -g @lightdash/cli`

on m1 mac having just run: `brew install npm`

### App Version

cloud eu

**Comment by ZeRego** [comment]
If possible, the cli should detect the OS and installs odbc.
If not, we add these instructions to the docs.

![Screenshot 2022-10-03 at 09 58 41](https://user-images.githubusercontent.com/9117144/193539061-42b0708c-a893-4444-afc1-c43c1941848a.png)

**Comment by rephus** [comment]
I've tried to add an error message if odbc drivers are not present, but I think it is potentially going to cause more issues. I managed to do the following:

Note that this error is happening on installation time (caused by odbc npm dependency), not on runtime. 

By adding a new `dependencies.sh` to npm `preinstall` script, I can check if one library file (eg: /etc/odbcinst.ini) is present, if not, I throw an error and display the dependency message.  But that file is going to be different on each OS, and I think this is going to be hard to test and maintain. I could also check the package on each OS, but again, it is hard to test. 

https://www.npmjs.com/package/odbc

![image](https://user-images.githubusercontent.com/1983672/193835675-26b4b3c3-3369-44a8-9062-1b1fe5d3bf8b.png)

@ZeRego what do you think ?

**Comment by ZeRego** [comment]
@rephus can we have that message as a warning instead of an error/blocker ? so If they get the error from odbc dependency they can find the warning.

**Comment by owlas** [comment]
Is there a chance of removing this? I think we added this because of the databricks connection. I believe that `dbt-databricks` can actually function without odbc. Not sure if our node drivers require it though

**Comment by ZeRego** [comment]
> Is there a chance of removing this? I think we added this because of the databricks connection. I believe that `dbt-databricks` can actually function without odbc. Not sure if our node drivers require it though

true, we could refactor completely the databrick warehouse client. Looks like they have a new npm package https://github.com/databricks/databricks-sql-nodejs that is still in beta. We currently use the spark driver, is this still necessary ?

**Comment by owlas** [comment]
Oh good find! I don't think the spark driver is necessary. For our own adapter I don't see any issues. 

The main concern is on the WarehouseCredentials <-> dbt profiles.yml conversion (both ways). For generating the `profiles.yml` (when hitting refresh dbt in the UI) we should be fine but I'm not sure about converting databricks profiles.yml from dbt back into WarehouseCredentials in the CLI
